deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -230,16 +235,71 @@ private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, f
 
     private <T> Optional<T> getValues(final Config config, final Path openApiFilePath, CodegenConfig.ConfigName configName,
             Class<T> propertyType) {
-        return config
-                .getOptionalValue(CodegenConfig.getSpecConfigName(configName, openApiFilePath), propertyType)
-                .or(() -> config.getOptionalValue(CodegenConfig.getGlobalConfigName(configName), propertyType));
+
+        Optional<String> possibleConfigKey = getConfigKeyValue(config, openApiFilePath);
+        boolean userConfiguredConfigKeyProperty = possibleConfigKey.isPresent() && !possibleConfigKey.get().isBlank();
+
+        if (userConfiguredConfigKeyProperty) {
+            return getValuesByConfigKey(config, CodegenConfig.getSpecConfigNameByConfigKey(possibleConfigKey.get(), configName),
+                    propertyType, configName);
+        }
+
+        return getValuesBySpecConfigName(config, openApiFilePath, configName, propertyType);
     }
 
     private <K, V> Optional<Map<K, V>> getValues(final SmallRyeConfig config, final Path openApiFilePath,
             CodegenConfig.ConfigName configName,
             Class<K> kClass, Class<V> vClass) {
+
+        Optional<String> possibleConfigKey = getConfigKeyValue(config, openApiFilePath);
+        boolean userConfiguredConfigKey = possibleConfigKey.isPresent() && !possibleConfigKey.get().isEmpty();
+
+        if (userConfiguredConfigKey) {
+            return getValuesByConfigKey(config, configName, kClass, vClass, possibleConfigKey.get());
+        }
```

Comment:
This code is pretty similar to what we have in lines 239-245. Maybe can we reuse it? Also, why you're checking for `isEmpty` there and `isBlank` here? Maybe we should also add unit tests for these configuration changes. wdyt?

Reply:
Hi @ricardozanini thank you for reviewing it. I think that it makes total sense!